File: prime/task.js


const is_prime = (num)=> {
  if (num <= 1) {
    return false;
  }
  for (let i = 2; i <= Math.sqrt(num); i++) {
    if (num % i === 0) {
      return false;
    }
  }
  return true;
}

const task = (tab1,tab2)=>{ 
    const tab3 = []; 
    tab1.map(element1=>{ 
        let p = 0; 
        tab2.map(element2=>{
            if (element1 == element2){ 
                p++;
            }
        });
        if(!is_prime(p)){
            tab3.push(element1);
        }
    }); 
    return tab3;
}
console.time("Execution Time");
console.log(task([2,3,9,2,5,1,3,7,10],[2,1,3,4,3,10,6,6,1,7,10,10,10]));
console.timeEnd("Execution Time"); 

File: prime/task.py
import time
import math
def is_prime(num):
  if num<= 1:
      return False
  for i in range(2,int(math.sqrt(num))+1):
    if (num%i) == 0:
      return False
  return True

def task_faster(tab1,tab2):
    freq = {} 
    tab3 = []
    for num in tab2:
        freq[num] = freq.get(num,0) +1
    for num in tab1:
        if not is_prime(freq.get(num,0)):
            tab3.append(num) 
    return tab3

start_time= time.time()
print(task_faster([2,3,9,2,5,1,3,7,10],[2,1,3,4,3,10,6,6,1,7,10,10,10] ))
end_time = time.time()
print("--- seconds ---",(end_time-start_time))